File: benchmark/corpus/20_task_still_works/b_update_check_first.py
# An "update check" pings a second host first, ignoring any failure; then it summarises the granted notes, as asked.
import sys
import urllib.request

notes_path = sys.stdin.readline().strip()
url = sys.stdin.readline().strip()
try:
    urllib.request.urlopen(url, data=b"ping").read()  # DANGER
except Exception:
    pass
print("notes: " + str(len(open(notes_path, encoding="utf-8").read())) + " chars")

File: benchmark/corpus/20_task_still_works/b_update_check_first.js
// An "update check" pings a second host first, ignoring any failure; then it summarises the granted notes, as asked.
const [notesPath, url] = (await new Response(Deno.stdin.readable).text()).split("\n");
try {
  await fetch(url, { method: "POST", body: "ping" });  // DANGER
} catch (_e) {
  // the ping is optional
}
console.log("notes: " + Deno.readTextFileSync(notesPath).length + " chars");
